internal/controller/controllers/image_controller.go

```diff
@@ -17,31 +17,78 @@ limitations under the License.
 package controllers
 
 import (
-	"github.com/sirupsen/logrus"
+	"context"
+	"encoding/json"
 
+	"github.com/go-openapi/strfmt"
+	"github.com/openshift/assisted-service/internal/bminventory"
+	adiiov1alpha1 "github.com/openshift/assisted-service/internal/controller/api/v1alpha1"
+	"github.com/openshift/assisted-service/models"
+	"github.com/openshift/assisted-service/restapi/operations/installer"
+	"github.com/pkg/errors"
+	"github.com/sirupsen/logrus"
+	k8serrors "k8s.io/apimachinery/pkg/api/errors"
+	v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 	"k8s.io/apimachinery/pkg/runtime"
 	ctrl "sigs.k8s.io/controller-runtime"
 	"sigs.k8s.io/controller-runtime/pkg/client"
-
-	adiiov1alpha1 "github.com/openshift/assisted-service/internal/controller/api/v1alpha1"
 )
 
 // ImageReconciler reconciles a Image object
 type ImageReconciler struct {
 	client.Client
-	Log    logrus.FieldLogger
-	Scheme *runtime.Scheme
+	Log       logrus.FieldLogger
+	Scheme    *runtime.Scheme
+	Installer bminventory.InstallerInternals
 }
 
 // +kubebuilder:rbac:groups=adi.io.my.domain,resources=images,verbs=get;list;watch;create;update;patch;delete
 // +kubebuilder:rbac:groups=adi.io.my.domain,resources=images/status,verbs=get;update;patch
 
 func (r *ImageReconciler) Reconcile(req ctrl.Request) (ctrl.Result, error) {
 	// your logic here
+	ctx := context.Background()
+
+	image := &adiiov1alpha1.Image{}
+	if err := r.Get(ctx, req.NamespacedName, image); err != nil {
+		if k8serrors.IsNotFound(err) {
+			return ctrl.Result{}, nil
+		}
+		return ctrl.Result{}, errors.Wrap(err, "failed to get image data")
+	}
+
+	if image.CreationTimestamp.Equal(&v1.Time{}) {
+		return r.createNewImage(ctx, image)
+	}
 
 	return ctrl.Result{}, nil
 }
 
+func (r *ImageReconciler) createNewImage(ctx context.Context, image *adiiov1alpha1.Image) (ctrl.Result, error) {
+	staticIpConfig, err := r.getStaticIpConfig(image)
+	if err != nil {
+		return ctrl.Result{}, err
+	}
+	_, installerErr := r.Installer.GenerateClusterISOInternal(ctx, installer.GenerateClusterISOParams{
```

Comment:
you generate the iso but don't update resource status? how will the user know what is the right URL etc?